src/NuGet.Core/NuGet.Packaging/Signing/Package/SignedPackageArchive.cs

```diff
@@ -70,6 +70,10 @@ public SignedPackageArchive(Stream packageReadStream, Stream packageWriteStream)
         {
             token.ThrowIfCancellationRequested();
 
+            // Don't update the read stream, do the removing in the write stream
+            ZipReadStream.Position = 0;
+            ZipReadStream.CopyTo(ZipWriteStream);
+
             if (ZipWriteStream == null)
```

Comment:
Minimally, this check should come before your `CopyTo(...)` statement [here](https://github.com/NuGet/NuGet.Client/pull/2051/files#diff-b91a030c7c291ec5eb1d411b8644bb13R75).  However, I'm pretty sure `ZipWriteStream` cannot be null, so this null check can probably be removed.